packages/gestalt/src/Masonry.tsx

```diff
@@ -622,18 +652,27 @@ export default class Masonry<T> extends ReactComponent<Props<T>, State<T>> {
       gridBody = (
         <div ref={this.setGridWrapperRef} style={{ width: '100%' }}>
           <div className={styles.Masonry} role="list" style={{ height, width }}>
-            {itemsToRender.map((item, i) =>
-              this.renderMasonryComponent(
-                item,
-                i,
-                // If we have items in the positionStore (newer way of tracking positions used for 2-col support), use that. Otherwise fall back to the classic way of tracking positions
-                // this is only required atm because the two column layout doesn't not return positions in their original item order
-                positionStore.get(item) ?? positions[i],
-              ),
-            )}
+            {itemsToRender.length === 0 && _loadingStateItems
+              ? _loadingStateItems.map((itemData, idx) =>
+                  this.renderMasonryComponent({
+                    itemData,
+                    idx,
+                    position: positionStore.get(itemData) ?? positions[idx],
+                    isLoading: true,
+                  }),
+                )
+              : itemsToRender.map((itemData, idx) =>
+                  this.renderMasonryComponent({
+                    itemData,
+                    idx,
+                    // If we have items in the positionStore (newer way of tracking positions used for 2-col support), use that. Otherwise fall back to the classic way of tracking positions
+                    // this is only required atm because the two column layout doesn't not return positions in their original item order
+                    position: positionStore.get(itemData) ?? positions[idx],
+                  }),
+                )}
           </div>
           <div className={styles.Masonry} style={{ width }}>
-            {itemsToMeasure.map((data, i) => {
+            {itemsToMeasure.map((data: T | U, i) => {
```

Comment:
same comment here but ideally we don't measure the loading state items since the loading state should be as cheap as possible to render